lua/neo-tree/ui/renderer.lua

```diff
@@ -938,7 +944,7 @@ local get_buffer = function(bufname, state)
     vim.api.nvim_buf_set_option(bufnr, "filetype", "neo-tree")
     vim.api.nvim_buf_set_option(bufnr, "modifiable", false)
     vim.api.nvim_buf_set_option(bufnr, "undolevels", -1)
-    autocmd.buf.define(bufnr, "WinLeave", function()
+    autocmd.buf.define(bufnr, "BufDelete", function()
```

Comment:
Why was this changed? Does it still work if the buffer is not being displayed in a window?

Reply:
When the buffer is not displayed in a window, users cannot change the position of the cursor right?

And for places that changes the cursor programatically (like follow current file), those cursor movements are captured in the _correct_ place (i.e. in `render_tree` or `close` etc).

Fun fact, that WinLeave _is_ the direct cause of the bug @mehalter was talking about in the issue.

Reply:
Makes sense.